cobigen-cli/cli/src/main/java/com/devonfw/cobigen/cli/commands/AdaptTemplatesCommand.java

```diff
@@ -0,0 +1,184 @@
+
+        // TODO: janv_capgemini check if sourcePath is an archive and throw exception if not
+        FileSystem fs = FileSystems.newFileSystem(sourcePath, null);
+
+        Path path = fs.getPath("/");
+        Files.walkFileTree(path, new SimpleFileVisitor<Path>() {
+            @Override
+            public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
+                Path relativePath = path.relativize(file);
+                Path targetPathResolved = targetPath.resolve(relativePath.toString());
+                Files.deleteIfExists(targetPathResolved);
+                Files.createDirectories(targetPathResolved.getParent());
+                Files.copy(file, targetPathResolved);
+                return FileVisitResult.CONTINUE;
+            }
+
+            @Override
+            public FileVisitResult visitFileFailed(Path file, IOException exc) throws IOException {
+                // Log errors but do not throw an exception
+                LOG.warn("An IOException occurred while reading a file on path {} with message: {}", file,
+                    exc.getMessage());
+                LOG.debug("An IOException occurred while reading a file on path {} with message: {}", file,
+                    LOG.isDebugEnabled() ? exc : null);
+                return FileVisitResult.CONTINUE;
+            }
+        });
+
+    }
+
+    @Override
+    public Integer call() throws Exception {
+        if (verbose) {
+            CLILogger.setLevel(Level.DEBUG);
+        }
+
+        LOG.info("Running AdaptTemplatesCommand!, please wait...");
+
+        Path cobigenTemplatesDirectory = null;
+        if (customTemplatesLocation != null) {
+            customTemplatesLocation = ConfigurationUtils.preprocessInputFile(customTemplatesLocation);
+            ConfigurationUtils.createConfigFile(customTemplatesLocation.toPath());
+            // sets custom templates directory path from configuration file property
+            cobigenTemplatesDirectory = ConfigurationUtils.getCustomTemplatesLocation();
+            LOG.info("Creating custom templates folder at custom location @ {}", cobigenTemplatesDirectory);
+        } else {
+            // sets default templates directory path from CLI location
+            cobigenTemplatesDirectory = ConfigurationUtils.getCobigenCliRootPath();
+            ConfigurationUtils.createConfigFile(cobigenTemplatesDirectory);
+            LOG.info("Creating custom templates folder next to the CLI @ {}", cobigenTemplatesDirectory);
+        }
+
+        if (Files.exists(cobigenTemplatesDirectory)) {
+            processJar(cobigenTemplatesDirectory);
+        } else {
+            LOG.error("Could not find target directory to extract templates @ {}", cobigenTemplatesDirectory);
+            return 0;
+        }
+
+        LOG.info("Successfully created custom templates folder @ {}",
+            cobigenTemplatesDirectory.resolve(ConfigurationUtils.COBIGEN_TEMPLATES));
```

Comment:
Currently we create a fixed CobiGen_Templates folder inside the custom location defined by the user. This message just shows the user that a new folder was created.

Reply:
Ok, I think, we should change that, meaning, that the user is providing a path including the project folder name and we extract templates to this folder. Please make sure, the command documentation is making this clear as well.
Why? - because there is no integration, which needs the CobiGen_Templates folder on the file path. The only thing, which needs "CobiGen_Templates" to exist, is the eclipse integration, but this is just related to the workspace model. In eclipse you could have a folder (file system), which is named as "A", whereas the `.project` file in the folder, could define the project in the workspace later on to be named like `CobiGen_Templates`, therefore, we don't have any restrictions on the folder name.

Also we want to get rid of the restriction within eclipse anyhow. So we should not introduce it here as well. But let's keep it as it is here for now and we will need to address this when also discussing where to put and configure the configuration folder in one of the next tasks.